linux/ibus-keyman/src/engine.c

```diff
@@ -745,21 +722,42 @@ process_capslock_action(km_core_caps_state caps_state) {
 
 static void
 commit_current_queue_item(IBusKeymanEngine *keyman) {
-  // only called for non-compliant apps with patched ibus
+  // only called for non-compliant apps
   g_assert(keyman != NULL);
-  g_assert(client_supports_prefilter((IBusEngine *)keyman));
-  g_assert(!client_supports_surrounding_text((IBusEngine *)keyman));
+  IBusEngine* engine = (IBusEngine *)keyman;
+  g_assert(!client_supports_surrounding_text(engine));
 
-  if (keyman->commit_item <= keyman->commit_queue)
+  if (keyman->commit_item <= keyman->commit_queue){
+    g_message("%s: queue is empty", __FUNCTION__);
     return;
+  }
 
   commit_queue_item *current_item = &keyman->commit_queue[0];
+  g_message("%s:", __FUNCTION__);
+  if (current_item->code_points_to_delete > 0) {
+    g_message("%s: Forwarding %d backspaces from commit queue", __FUNCTION__, current_item->code_points_to_delete);
+    while (current_item->code_points_to_delete > 0) {
+      ibus_engine_forward_key_event(engine, KEYMAN_BACKSPACE_KEYSYM, KEYMAN_BACKSPACE, 0);
+      current_item->code_points_to_delete--;
+    }
+    // don't remove the item from the queue yet - we need to process it
+    // again for the output and keystrokes. Instead emit the sentinel key
+    // again.
+    g_message("%s: Forcing ordered output", __FUNCTION__);
+    call_ordered_output_sentinel();
+    return;
+  }
   if (current_item->char_buffer != NULL) {
+    g_autofree gchar *debug = NULL;
+    g_message("%s: Committing from commit queue: %s", __FUNCTION__,
+      debug = debug_utf8_with_codepoints(current_item->char_buffer));
```

Comment:
Is this temporary logging of the character buffer?

Reply:
This is debug output that is intended to stay. However, we should change it to use `g_debug` instead of `g_message`. I'll write up an issue for that.